docs/advanced/integration_tests.md

```diff
@@ -0,0 +1,132 @@
+
+### ApiServer integration tests
+
+The _client_ object is permissive, allowing access to the request itself. A simpler way to test an API is provided by Gosoline's _ApiServerTestCase_. These type of tests also issue HTTP calls and compare their responses against a predefined value, but they need to follow a predefined structure. Below we can see how the _resty client_ test above would look like as an _ApiServerTestCase_:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) Test_Euro/ /\n}/)
+```go
+func (s *ApiTestSuite) Test_Euro() *suite.ApiServerTestCase {
+	return &suite.ApiServerTestCase{
+		Method:             http.MethodGet,
+		Url:                "/euro/10/GBP",
+		Headers:            map[string]string{},
+		ExpectedStatusCode: http.StatusOK,
+		Assert: func(response *resty.Response) error {
+			s.Equal(http.StatusOK, response.StatusCode())
+
+			result, err := strconv.ParseFloat(string(response.Body()), 64)
+			s.NoError(err)
+			s.Equal(8.0, result)
+
+			return nil
+		},
+	}
+}
+```
+
+Notice the predefined structure of an _ApiServerTestCase_: it is an object that has fields for all the information needed in performing an HTTP call, including a method that tests the result for correctness.
+
+In the example from `examples/getting_started/integration/api_test.go` we can see that both types of test cases (_resty.Client_ and _ApiServerTestCase_) share the same `SetupSuite` method.
+
+### Stream integration tests
+
+If you have a Gosoline application that takes its input from a stream, you need a test which can run your application locally, send data to a stream, and check your application's output for correctness. 
+
+The `stream-consumer` application, found in `advanced/stream-consumer`, reads unsigned integers from an `inputEvent` sqs queue, increments them by one and publishes them to the `publisher-outputEvent` SNS topic. Both the SQS queue which provides it with input, and the SNS topic to which it publishes its results, are configured in its `config.dist.yml` file. While at runtime its input is an actual SNS topic and its output an SQS queue, while running the integration tests those will actually be in memory inputs/outputs. Below is an extract from its integration test:
+
+[embedmd]:# (../../examples/advanced/stream-consumer-test/stream_consumer_test.go /func \(s \*ConsumerTestSuite\) SetupSuite/ /\n}/)
+```go
+func (s *ConsumerTestSuite) SetupSuite() []suite.Option {
+	return []suite.Option{
+		suite.WithLogLevel("debug"),
+		suite.WithConfigFile("../stream-consumer/config.dist.yml"),
+		suite.WithModule("consumer", stream.NewConsumer("consumer", consumer.NewConsumer())),
+	}
+}
+```
+
+It is making use of the same `config.dist.yml` file as `stream-consumer`, and it will be using the module created by _consumer.NewConsumer_. The difference between a _StreamTestCase_ test and an _ApiServerTestCase_ can be seen in its test cases:
```

Comment:
I would single out the `WithModule` line and not even show the other lines from `SetupSuite`. The important part here is that for an API server we *don't* create the module for the kernel, for a consumer we do. The whole log level & config file stuff is just noise we have seen a thousand times at this point